ivy/functional/backends/jax/device.py

```diff
@@ -75,13 +75,22 @@ def _to_device(x, device=None):
 
 def as_ivy_dev(device, /):
     if isinstance(device, str):
-        return ivy.Device(device)
-    if device is None:
+        device_object = ivy.Device(device)
+    elif device is None:
         return None
-    p, dev_id = (device.platform, device.id)
-    if p == "cpu":
-        return ivy.Device(p)
-    return ivy.Device(p + ":" + str(dev_id))
+    else:
+        p, dev_id = (device.platform, device.id)
+        if p == "cpu":
+            device_object = ivy.Device(p)
+        else:
+            device_object = ivy.Device(p + ":" + str(dev_id))
+    
+    # Verify if the device exists on the host
+    available_devices = jax.local_devices()
```

Comment:
Actually, the issue here is that `jax.local_devices()` returns `jax` native devices (e.g. `CpuDevice(id=0)` while we're trying to compare it against an `ivy.Device` which was created on line [86](https://github.com/unifyai/ivy/pull/18247/files#diff-2b01bb1c1257260fd336443a518ea5713550f12a81ba35ff6ff08414f7e6f3c8R86) (and which subclasses `str`). As a result, the result of `str(device_object) not in available_devices` is always `False` leading to [this](https://github.com/unifyai/ivy/actions/runs/5535639911/jobs/10106420561?pr=18247#step:3:31) error.
Basically we'd need to convert the `available_devices` list to instead have elements of the form `f'{device.platform}:{device.id}` if `device.platform` isn't a `cpu` and use `f'{device.platform}'` otherwise,
```
available_devices = [f'{device.platform}:{device.id}' if device.platform != 'cpu' else device.platform for device in available_devices]
```
Once we've done that, we'd be able to correctly do a `str(device_object) not in available_devices` 
Thanks @sladyn98 😄👍